packages/react-i18n-webpack-plugin/src/react-i18n-webpack-plugin.ts

```diff
@@ -0,0 +1,264 @@
+import path from 'path';
+
+import webpack from 'webpack';
+import {camelCase} from 'change-case';
+import VirtualModulesPlugin from 'webpack-virtual-modules';
+import ParserHelpers from 'webpack/lib/ParserHelpers';
+import {CallExpression, Expression, SpreadElement} from 'estree';
+
+import {FindI18nImportPlugin} from './find-i18n-imports-plugin';
+import {generateID} from './utilities';
+
+const PLUGIN_NAME = 'ReactI18nPlugin';
+const TRANSLATION_DIRECTORY_NAME = 'translations';
+
+export interface Options {
+  fallbackLocale: string;
+}
+
+export class ReactI18nPlugin {
+  private options: Options;
+
+  private defaultOptions: Options = {
+    fallbackLocale: 'en',
+  };
+
+  constructor(options: Partial<Options> = {}) {
+    this.options = {
+      ...this.defaultOptions,
+      ...options,
+    };
+  }
+
+  apply(compiler: webpack.Compiler) {
+    const virtualModules = new VirtualModulesPlugin();
+    virtualModules.apply(compiler);
+
+    compiler.hooks.compilation.tap(
+      PLUGIN_NAME,
+      (
+        compilation: webpack.compilation.Compilation,
+        {
+          normalModuleFactory,
+        }: {normalModuleFactory: webpack.compilation.NormalModuleFactory},
+      ) => {
+        const handler = (parser: any) => {
+          new FindI18nImportPlugin().apply(parser);
```

Comment:
I find it way more confusing to have this as a separate plugin, personally

Reply:
😂

Reply:
Confusing for me, too.  If we're trying to slim down the body of this function, there are better parts to refactor into helper functions.

Reply:
ok, moving it back in!